### PR TITLE
Replace guava ThreadFactoryBuilder with micrometer NamedThreadFactory

### LINKED ISSUE #3086: Zipkin-slim complains "NoClassDefFoundError" when setting up ES_CREDENTIALS_FILE
## Describe the Bug
When setting up ES_CREDENTIALS_FILE to enable credentials file for `zipkin-slim`, a `java.lang.NoClassDefFoundError: com/google/common/util/concurrent/ThreadFactoryBuilder` is raised. The exception is from https://github.com/openzipkin/zipkin/blob/3aa6695bfdee6d0352a589ee10355b0fde705861/zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageConfiguration.java#L166
which leverages guava `ThreadFactoryBuilder` to build a thread factory for credentials file refresher.

I found that's because`guava` is excluded from slim. 
https://github.com/openzipkin/zipkin/blob/3aa6695bfdee6d0352a589ee10355b0fde705861/zipkin-server/pom.xml#L510

There seem two paths to fix it.

1. Remove ThreadFactoryBuilder to get rid of guava dependency.
1. Add guava back to slim.

I prefer the first approach. @adriancole WDYT?

### PR DESCRIPTION
**Pull request by hanahmily**
Fixed #3086 

Signed-off-by: Gao Hongtao <hanahmily@gmail.com>

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
thanks

File: zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageConfiguration.java [hunk 1]
```diff
@@ -163,9 +163,7 @@ ScheduledExecutorService dynamicCredentialsScheduledExecutorService(
     @Value("${" + CREDENTIALS_REFRESH_INTERVAL + "}") Integer credentialsRefreshInterval,
     @Qualifier(QUALIFIER) BasicCredentials basicCredentials) throws IOException {
     ScheduledExecutorService ses = Executors.newSingleThreadScheduledExecutor(
-      new ThreadFactoryBuilder()
-        .setNameFormat("LoadElasticSearchCredentials-%d")
-        .build());
+      new NamedThreadFactory("zipkin-load-es-credentials-pool"));
```
Comment (jorgheymans) [inline zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageConfiguration.java]: Minor nit: drop the pool suffix perhaps.

Comment (hanahmily) [reply zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageConfiguration.java]: done

**Comment by jorgheymans** [comment]
Thanks @hanahmily , looking forward to your next PR ;-)

**Pull request merged**